test/randomized/test_synthesis.py

```diff
@@ -0,0 +1,155 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Randomized tests of quantum synthesis."""
+import unittest
+from test.python.quantum_info.test_synthesis import CheckDecompositions
+from hypothesis import given, strategies
+import numpy as np
+
+from qiskit import execute
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.extensions import UnitaryGate
+from qiskit.providers.basicaer import UnitarySimulatorPy
+from qiskit.quantum_info.random import random_unitary
+from qiskit.quantum_info.synthesis.two_qubit_decompose import (two_qubit_cnot_decompose,
+                                                               TwoQubitBasisDecomposer,
+                                                               Ud)
+
+
+class TestSynthesis(CheckDecompositions):
+    """Test synthesis"""
+
+    @given(strategies.integers(min_value=0, max_value=2 ** 32 - 1))
+    def test_1q_random(self, seed):
+        """Checks one qubit decompositions"""
+        unitary = random_unitary(2, seed=seed)
+        self.check_one_qubit_euler_angles(unitary)
+        self.check_one_qubit_euler_angles(unitary, 'U3')
+        self.check_one_qubit_euler_angles(unitary, 'U1X')
+        self.check_one_qubit_euler_angles(unitary, 'ZYZ')
+        self.check_one_qubit_euler_angles(unitary, 'ZXZ')
+        self.check_one_qubit_euler_angles(unitary, 'XYX')
+
+    @given(strategies.integers(min_value=0, max_value=2 ** 32 - 1))
+    def test_2q_random(self, seed):
+        """Checks two qubit decompositions"""
+        unitary = random_unitary(4, seed=seed)
+        self.check_exact_decomposition(unitary.data, two_qubit_cnot_decompose)
+
+    @given(strategies.tuples(strategies.integers(min_value=0, max_value=2 ** 32 - 1),
+                             strategies.integers(min_value=0, max_value=2 ** 32 - 1),
+                             strategies.integers(min_value=0, max_value=2 ** 32 - 1),
+                             strategies.integers(min_value=0, max_value=2 ** 32 - 1),
+                             strategies.integers(min_value=0, max_value=2 ** 32 - 1)))
+    def test_exact_supercontrolled_decompose_random(self, seeds):
+        """Exact decomposition for random supercontrolled basis and random target"""
+        # pylint: disable=invalid-name
+        k1 = np.kron(random_unitary(2, seed=seeds[0]).data, random_unitary(2, seed=seeds[1]).data)
+        k2 = np.kron(random_unitary(2, seed=seeds[2]).data, random_unitary(2, seed=seeds[3]).data)
+        basis_unitary = k1 @ Ud(np.pi / 4, 0, 0) @ k2
+        decomposer = TwoQubitBasisDecomposer(UnitaryGate(basis_unitary))
+        self.check_exact_decomposition(random_unitary(4, seed=seeds[4]).data, decomposer)
+
+    rotation = strategies.floats(min_value=-np.pi*10, max_value=np.pi*10)
```

Comment:
Not fully sure if I understand correctly. You mean for the `execute`? Something like this 3b6d4ccdc ?

Reply:
Using it for a simulator seed is a good idea, I had meant using it to consolidate the `givens` on L34, L45, and L51-55.

Reply:
consolidating wound that reduce the space?

Reply:
The search space would be the same, this would just avoid having to redefine the integer strategy. e.g. instead of
```
    @given(strategies.tuples(strategies.integers(min_value=0, max_value=2 ** 32 - 1),
                             strategies.integers(min_value=0, max_value=2 ** 32 - 1),
                             strategies.integers(min_value=0, max_value=2 ** 32 - 1),
                             strategies.integers(min_value=0, max_value=2 ** 32 - 1),
                             strategies.integers(min_value=0, max_value=2 ** 32 - 1)))
```
you could
```
seed = strategies.integers(min_value=0, max_value=2 ** 32 - 1)
...
@given(strategies.tuples(*[seed]*5)
```

Reply:
oh! I see. Done in 904c23c43